src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -818,6 +818,14 @@ IndexerConnector::IndexerConnector(
                 {
                     m_db->delete_(id);
                 }
+                // We made the same operation for DELETED_BY_QUERY as for DELETED
+                else if (parsedData.at("operation").get_ref<const std::string&>().compare("DELETED_BY_QUERY") == 0)
+                {
+                    for (const auto& [key, _] : m_db->seek(id))
+                    {
+                        m_db->delete_(key);
+                    }
+                }
```

Comment:
The changes are Ok.

I'd just add the check for `data` field below, now that we are aligning both methods.
Moving the duplicate code to a common function might require more changes, so we need to replicate all cases.

Reply:
Done! Thanks